### PR TITLE
[added] ExternalStream to StreamSourceInfo and updated server

### PR DESCRIPTION
**Pull request by matthiashanel**
Instead of adding the field the server struct itself is used.

Signed-off-by: Matthias Hanel <mh@synadia.com>

Reflecting this [change](https://github.com/nats-io/nats-server/commit/060f2ec127d231d098c171e7d180925b336d5382) in jsml.

Screen Shot 2021-05-14 at 1 23 31 PM
Screen Shot 2021-05-14 at 1 23 23 PM

### CODE REVIEW

**Review by ripienaar — CHANGES_REQUESTED** [review]
Please add locally.

**Comment by matthiashanel** [comment]
Added locally instead

File: api/streams.go [hunk 1]
```diff
@@ -362,10 +362,11 @@ type Placement struct {
 
 // StreamSourceInfo shows information about an upstream stream source.
 type StreamSourceInfo struct {
-	Name   string        `json:"name"`
-	Lag    uint64        `json:"lag"`
-	Active time.Duration `json:"active"`
-	Error  *ApiError     `json:"error,omitempty"`
+	Name     string          `json:"name"`
+	External *ExternalStream `json:"external,omitempty"`
```
Comment (ripienaar) [inline api/streams.go:366]: We need to add this to the schemas as well

**Comment by ripienaar** [comment]
LGTM

**Pull request merged**